tests/modules.rs

```diff
@@ -685,6 +685,47 @@ fn module_paths_beginning_with_tilde_are_expanded_to_homdir() {
     .run();
 }
 
+#[test]
+fn module_paths_with_variables_are_expanded() {
+  Test::new()
+    .write("foobar/mod.just", "foo:\n @echo FOOBAR")
+    .justfile(
+      "
+        mod foo '$FOO/mod.just'
+      ",
+    )
+    .test_round_trip(false)
+    .arg("--unstable")
+    .arg("foo")
+    .arg("foo")
+    .stdout("FOOBAR\n")
+    .env("FOO", "foobar")
+    .run();
+}
+
+#[test]
+fn unresolved_variable_in_module_error() {
+  Test::new()
+    .justfile(
+      "
+        mod foo '$FOO/mod.just'
+      ",
+    )
+    .test_round_trip(false)
+    .arg("--unstable")
+    .status(EXIT_FAILURE)
+    .stderr(
+      "
+      error: Source path for foo contains unresolved variable FOO.
+       ——▶ justfile:1:5
+        │
+      1 │ mod foo '$FOO/mod.just'
+        │     ^^^
```

Comment:
The pointers here would be nicer on the path, but I couldn't figure out how to get to a token from `relative`

Reply:
Relative is an Option<StringLiteral>, but in the case of an error, you know there's a path, so you can unwrap it.

Reply:
It's already unwrapped. The difficulty I face is that `relative` is a `StringLiteral`, and I need a `Token` in the error handling to position the pointers. I don't know how to get back from the literal string to the token.
I thought about perhaps using the module token and iterating to the next token after (something like `module.token.next`), but I don't see an easy way to do it this way either :confused:

Reply:
@casey Any remaining ideas on this? Otherwise I keep it as is :slightly_frowning_face:

Reply:
The `mod` parser uses `parse_string_literal`, but there's also `parse_string_literal` token which returns both the string literal and the token, if you store the token in `mod` in addition to the string literal, you can use it in the error message.